### DISCUSSION TITLE
How to adapt size of AxisTicks in R

### DISCUSSION THREAD

**Discussion by nina-nauwelaerts** [Open-Systems-Pharmacology/Forum#1875] [Questions & Problems]
I would like to increase the size of the numbers on the x- and y-axis on a plot using the ospsuite-R package. Is this possible using the PlotConfiguration?

**Comment by pchelle** [comment]
@nina-nauwelaerts , it is indeed possible by defining [`DefaultPlotConfiguration`](https://www.open-systems-pharmacology.org/OSPSuite-R/reference/DefaultPlotConfiguration.html#public-fields) objects:

``` r
plotConfig <- DefaultPlotConfiguration$new()
plotConfig$xAxisLabelTicksSize <- 16
plotConfig$yAxisLabelTicksSize <- 16

plotIndividualTimeProfile(myDataCombined, plotConfig)
```




<details><summary>Here is an example taken from the documentation</summary>

``` r
library(ospsuite)
#> Loading required package: rSharp

# simulated data
simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
sim <- loadSimulation(simFilePath)
simResults <- runSimulations(sim)[[1]]
outputPath <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"

# observed data
obsData <- lapply(
  c("ObsDataAciclovir_1.pkml", "ObsDataAciclovir_2.pkml", "ObsDataAciclovir_3.pkml"),
  function(x) loadDataSetFromPKML(system.file("extdata", x, package = "ospsuite"))
)
names(obsData) <- lapply(obsData, function(x) x$name)

myDataCombined <- DataCombined$new()

myDataCombined$addSimulationResults(
  simulationResults = simResults,
  quantitiesOrPaths = outputPath,
  groups = "Aciclovir PVB"
)

myDataCombined$addDataSets(
  obsData$`Vergin 1995.Iv`,
  groups = "Aciclovir PVB"
)

plotIndividualTimeProfile(myDataCombined)
```

![](https://i.imgur.com/bfiW6qk.png)

``` r
plotConfig <- DefaultPlotConfiguration$new()
plotConfig$xAxisLabelTicksSize <- 16
plotConfig$yAxisLabelTicksSize <- 16

plotIndividualTimeProfile(myDataCombined, plotConfig)
```

![](https://i.imgur.com/wolm5S4.png)

<sup>Created on 2024-12-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

</details>

**Reply by nina-nauwelaerts** [reply]
Thank you for your reply. I have tried to change the size as you indicate, but it doesn't seem to work. I do not have any problem changing the size of the title or axis label, but I can't change the size of the numbers. I can see the size changing in the plot configuration, but the plot remains the same.

**Reply by pchelle** [reply]
Could you let me know which version of `{ospsuite}` you are using ?
It seems there was such an issue at some point [see here](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1184) which has been solved since then. Not sure if fixed in the [current release](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/releases/tag/v11.3.520) though.

**Reply by pchelle** [reply]
If you do not want to change your version of `{ospsuite}`, you can also directly do it through `{ggplot2}`:

```r
plotIndividualTimeProfile(myDataCombined) + 
    ggplot2::theme(
        axis.text.x = ggplot2::element_text(size = 16),
        axis.text.y = ggplot2::element_text(size = 16)
        )
```

**Reply by nina-nauwelaerts** [reply]
I'm sorry I missed that issue, I am indeed using an older version (11.3.520) so that explains the problem.
